aspnetcore/tutorials/web-api-vsc.md

```diff
@@ -0,0 +1,272 @@
+
+This defines an empty controller class. In the next sections, we'll add methods to implement the API.
+
+## Getting to-do items
+
+To get to-do items, add the following methods to the `TodoController` class.
+
+[!code-csharp[Main](first-web-api/sample/TodoApi/Controllers/TodoController.cs?name=snippet_GetAll)]
+
+These methods implement the two GET methods:
+
+* `GET /api/todo`
+
+* `GET /api/todo/{id}`
+
+Here is an example HTTP response for the `GetAll` method:
+
+```
+HTTP/1.1 200 OK
+   Content-Type: application/json; charset=utf-8
+   Server: Microsoft-IIS/10.0
+   Date: Thu, 18 Jun 2015 20:51:10 GMT
+   Content-Length: 82
+
+   [{"Key":"4f67d7c5-a2a9-4aae-b030-16003dd829ae","Name":"Item1","IsComplete":false}]
+   ```
+
+Later in the tutorial I'll show how you can view the HTTP response using [Postman](https://www.getpostman.com/).
+
+### Routing and URL paths
+
+The `[HttpGet]` attribute specifies an HTTP GET method. The URL path for each method is constructed as follows:
+
+* Take the template string in the controller’s route attribute,  `[Route("api/[controller]")]`
+* Replace "[Controller]" with the name of the controller, which is the controller class name minus the "Controller" suffix. For this sample, the controller class name is **Todo**Controller and the root name is "todo". ASP.NET Core [routing](xref:mvc/controllers/routing) is not case sensitive.
+* If the `[HttpGet]` attribute has a template string, append that to the path. This sample doesn't use a template string.
+
+In the `GetById` method:
+
+```csharp
+[HttpGet("{id}", Name = "GetTodo")]
+public IActionResult GetById(string id)
+```
+
+`"{id}"` is a placeholder variable for the ID of the `todo` item. When `GetById` is invoked, it assigns the value of "{id}" in the URL to the method's `id` parameter.
+
+`Name = "GetTodo"` creates a named route and allows you to link to this route in an HTTP Response. I'll explain it with an example later. See [Routing to Controller Actions](xref:mvc/controllers/routing) for detailed information.
+
+### Return values
+
+The `GetAll` method returns an `IEnumerable`. MVC automatically serializes the object to [JSON](http://www.json.org/) and writes the JSON into the body of the response message. The response code for this method is 200, assuming there are no unhandled exceptions. (Unhandled exceptions are translated into 5xx errors.)
+
+In contrast, the `GetById` method returns the more general `IActionResult` type, which represents a wide range of return types. `GetById` has two different return types:
+
+* If no item matches the requested ID, the method returns a 404 error.  This is done by returning `NotFound`.
+
+* Otherwise, the method returns 200 with a JSON response body. This is done by returning an `ObjectResult`
+
+  
+### Launch the app
+
+In VS Code, press CTRL+F5 to launch the app. Navigate to  http://localhost:port/api/todo   (The `Todo` controller we just created).
```

Comment:
launch the app ... in a browser

could replace "port" with 5000 since we know what it is in this tutorial

Stuck at this point ... "Unable to resolve service for type 'TodoApi.Models.TodoContext' while attempting to activate 'TodoApi.Models.TodoRepository'."  --  on the earlier snippet for Startup.cs you're missing the highlighting on this line:

services.AddDbContext<TodoContext>(opt => opt.UseInMemoryDatabase());

Also need a using there

using Microsoft.EntityFrameworkCore;